Sources/ApodiniObserveOpenTelemetry/InstrumentConfiguration+OpenTelemetry.swift

```diff
@@ -0,0 +1,64 @@
+//
+// This source file is part of the Apodini open source project
+//
+// SPDX-FileCopyrightText: 2019-2021 Paul Schmiedmayer and the Apodini project authors (see CONTRIBUTORS.md) <paul.schmiedmayer@tum.de>
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Apodini
+import ApodiniObserve
+import OpenTelemetry
+import OtlpGRPCSpanExporting
+import Tracing
+
+/// Extension that allows for easier setup of the ``InstrumentConfiguration`` for an OpenTelemetry backend.
+extension InstrumentConfiguration {
+    /// Default OpenTelemetry `Instrument`, exporting spans to an OpenTelemetry collector via gRPC in the OpenTelemetry protocol (OTLP).
+    /// - Parameter serviceName: The name of the `WebService` being traced.
+    public static func defaultOpenTelemetry(serviceName: String) -> InstrumentConfiguration {
+        openTelemetryWithConfig(
+            serviceName: serviceName,
```

Comment:
It would be cool if we could somehow get a default value for this from Apodini, though I didn't really find anything appropriate.

Reply:
A thing that would come to my mind would be the `hostname ` in the `HTTPConfiguration`: https://github.com/Apodini/Apodini/blob/fac5afd188640b794ec3dda6283d873ea6dd59d0/Sources/Apodini/Configurations/HTTPConfiguration.swift . It should ideally be set to e.g. `ase.in.tum.de`. The only tricky thing is that the InstrumentConfiguration could come after the HTTP configuration and therefore the name might not by populated at this point. But this is something that might not be easily solvable and is something that we should highlight in the documentation that the order is relevant.

Reply:
A thing that would come to my mind would be the `hostname ` in the `HTTPConfiguration`: https://github.com/Apodini/Apodini/blob/fac5afd188640b794ec3dda6283d873ea6dd59d0/Sources/Apodini/Configurations/HTTPConfiguration.swift . It should ideally be set to e.g. `ase.in.tum.de`. The only tricky thing is that the InstrumentConfiguration could come after the HTTP configuration and therefore the name might not by populated at this point. But this is something that might not be easily solvable and is something that we should highlight in the documentation that the order is relevant.

Reply:
Hmm, right.. I'll mention it in the docs and try to populate the name from the hostname when it's available.

Also, it might be interesting to think about a mechanism to enforce the order of some configurations because for example [Apple also recommends to bootstrap Logging, Metrics, and Tracing in this specific order](https://github.com/apple/swift-distributed-tracing#recommended-bootstrap-order).

Reply:
Is the serviceName here the same the hostname, though? I'd imagine the "name" of the web service to be the product (e.g. "Xpense") rather than the hostname on which it just so happens to reside (e.g. xpense.ast.in.tum.de).

This is actually something I've been thinking about a bit, since we currently have several places where the "name" of the web service being exported is somehow used/required. The first, iirc was the OpenAPIInterfaceExporter (bc the OpenAPI definition contains a name field). The gRPC IE also has configuration options for "package name" and "service name" (the first being protobuf-specific). And GraphQL might end up needing something like this as well.

It'd probably be a good idea to have a top-level (i.e. not belonging to any specific IE) configuration option for specifying these kinds of metadata about a web service. This would need to come at the very beginning of the configuration, and then make this available to all other interface exporters.

(If your comments were in fact about the actual hostname, rather than the end-user-legible name of the service, I guess you can ignore my comment here...)

Reply:
Right, I agree with Lukas here. The service names in the tracing context tend to be something like "frontend", or "customer". That's why I initially thought about using the actual name of the type that is implementing the `WebService` protocol.